### PR TITLE
Popup menu search

### PR DESCRIPTION
**Pull request by philippfromme**
* adds ability for popup menu providers to provide custom search
* default search sorts matches by location (when searching for "foo", "foo bar" will appear before "bar foo")

### CODE REVIEW

**Comment by philippfromme** [comment]
Example usage here: https://github.com/camunda/camunda-bpmn-js/pull/367/files#diff-1abbb6d8413b72aac12009eae34de187cdc46f9c9eeb5daad527e5c7c43a8df8R23

Not entirely shure if this is the way to do it or wether we should have separate providers for search similar to: https://github.com/bpmn-io/diagram-js/blob/develop/lib/features/search-pad/SearchPadProvider.ts

File: lib/features/popup-menu/PopupMenu.js [hunk 1]
```diff
@@ -557,6 +570,15 @@ PopupMenu.prototype._getEmptyPlaceholder = function(providers) {
 };
 
 
+PopupMenu.prototype._getFindEntries = function(providers) {
+  const provider = providers.find(
```
[the remark this answers is not in the document]
Comment (philippfromme) [reply lib/features/popup-menu/PopupMenu.js:574]: So the idea would be for both the popup menu and the global search to do:

```javascript

// search available through injector, can be overridden, similar to `translate`
const searchFn = injector.get('search');
```

What do you think?

[the remark this answers is not in the document]
Comment (nikku) [reply lib/features/popup-menu/PopupMenu.js:574]: Yes, this is what I'm thinking.

[the remark this answers is not in the document]
Comment (nikku) [reply lib/features/popup-menu/PopupMenu.js:574]: I'm happy to sketch this next week.

[the remark this answers is not in the document]
Comment (philippfromme) [reply lib/features/popup-menu/PopupMenu.js:574]: Okay, I think this could be a good direction. 👍🏻

**Review by nikku — APPROVED** [review]
Getting back to this PR after https://github.com/bpmn-io/diagram-js/pull/929 is merged. I think this is great!

image

**Comment by nikku** [comment]
Released as [diagram-js@v14.11.0](https://github.com/bpmn-io/diagram-js/releases/tag/v14.11.0).

**Comment by nikku** [comment]
Example integration surfaced some issues that we may want to follow-up on: https://github.com/bpmn-io/bpmn-js/pull/2235.

**Pull request merged**